apps/src/dance/blockly/blocks.js

```diff
@@ -76,6 +79,51 @@ const customInputTypes = {
       return block.getFieldValue(arg.name);
     },
   },
+  dancerPicker: {
+    addInput(blockly, block, inputConfig, currentInputRow) {
+      const options = inputConfig.options.map(option => {
+        let name = option;
+        try {
+          name = JSON.parse(name);
+        } catch {}
+        if (name === GENERATED_DANCER) {
+          const localStorageDancer = JSON.parse(
+            localStorage.getItem('dancer-ai-generate')
+          );
+          const {adlibOption, choices, variant} = localStorageDancer || {};
```

Comment:
Do we have any way to abstract out how we fetch the dancer metadata? For both dance and music, we will need a more robust solution than just fetching from local storage to avoid overwriting previous work, making sure users aren't getting other users' data on the same machine, etc. Likely that data will involve some sort of lookup based on level ID/channel ID, which means it will have to happen after the level has loaded. Could we use a mutation/extension here potentially to load in the dancer data after the level has loaded?

In any case, probably fine to ship this for now for playtesting, but let's add a comment/TODO here?